studio/components/interfaces/Settings/General/CustomDomainConfig/CustomDomainsConfigureHostname.tsx

```diff
@@ -14,30 +14,48 @@ import {
 } from 'components/ui/Forms'
 import { useCustomDomainCreateMutation } from 'data/custom-domains/custom-domains-create-mutation'
 
-const FORM_ID = 'custom-domains-form'
-
 const schema = yup.object({
   domain: yup.string().required('A value for your custom domain is required'),
 })
 
-export type CustomDomainsConfigureHostnameProps = {
-  projectRef?: string
-}
-
-const CustomDomainsConfigureHostname = ({ projectRef }: CustomDomainsConfigureHostnameProps) => {
+const CustomDomainsConfigureHostname = () => {
   const { ui } = useStore()
+  const { ref } = useParams()
+  const { mutateAsync: createCustomDomain } = useCustomDomainCreateMutation()
+  const { services } = useProjectSettings(ref as string | undefined)
+
+  // Get the API service
+  const API_SERVICE_ID = 1
+  const apiService = services ? services.find((x: any) => x.app.id == API_SERVICE_ID) : {}
+  const endpoint = apiService?.app_config?.endpoint
 
+  const FORM_ID = 'custom-domains-form'
   const canConfigureCustomDomain = checkPermissions(PermissionAction.UPDATE, 'projects')
-  const { mutateAsync: createCustomDomain } = useCustomDomainCreateMutation()
+
+  const verifyCNAME = async (domain: string): Promise<boolean> => {
+    const res = await fetch(`https://1.1.1.1/dns-query?name=${domain}`, {
+      method: 'GET',
+      headers: { accept: 'application/dns-json' },
+    })
+    const verification = await res.json()
+    return verification.Status === 0
+  }
 
   const onCreateCustomDomain = async (values: yup.InferType<typeof schema>) => {
-    if (!projectRef) {
-      throw new Error('Project ref is required')
+    if (!ref) throw new Error('Project ref is required')
+
+    const cnameVerified = await verifyCNAME(values.domain)
+    if (!cnameVerified) {
+      return ui.setNotification({
+        category: 'error',
+        duration: 4000,
+        message: `Your CNAME record for ${values.domain} has yet to be verified - please check back again in a bit.`,
```

Comment:
`has yet to be verified` seems to be imply that the issue is on our end, whereas in reality they might not have set the CNAME record as expected.

We should say we expect it to be X, but it was Y (or missing). Also, would not auto-dismiss the error notification.

Reply:
gotcha - updated the language here